tests/std/tests/GH_005421_vector_algorithms_integer_class_type_iterator/test.cpp

```diff
@@ -0,0 +1,223 @@
+
+    assert(ranges::min_element(arr_begin, arr_end) == arr_begin);
+    assert(ranges::max_element(arr_begin, arr_end) == arr_end - _Signed128{1});
+    assert(ranges::minmax_element(arr_begin, arr_end).max == arr_end - _Signed128{1});
+
+    assert(ranges::min(ranges::subrange(arr_begin, arr_end)) == 200);
+    assert(ranges::max(ranges::subrange(arr_begin, arr_end)) == 390);
+    assert(ranges::minmax(ranges::subrange(arr_begin, arr_end)).max == 390);
+
+    {
+        // floating minmax is distinct codepath unless /fp:fast
+        float float_arr[arr_size];
+        picky_contiguous_iterator float_arr_begin(begin(float_arr));
+        picky_contiguous_iterator float_arr_end(end(float_arr));
+
+        transform(arr_begin, arr_end, float_arr_begin, [](int v) { return static_cast<float>(v); });
+
+        assert(ranges::min(ranges::subrange(float_arr_begin, float_arr_end)) == 200.0);
+        assert(ranges::max(ranges::subrange(float_arr_begin, float_arr_end)) == 390.0);
+        assert(ranges::minmax(ranges::subrange(float_arr_begin, float_arr_end)).max == 390.0);
+    }
+
+    assert(is_sorted_until(arr_begin, arr_end) == arr_begin + _Signed128{4});
+    assert(ranges::is_sorted_until(arr_begin, arr_end) == arr_begin + _Signed128{4});
+
+    {
+        int arr_copy[arr_size];
+        picky_contiguous_iterator arr_copy_begin(begin(arr_copy));
+        picky_contiguous_iterator arr_copy_end(end(arr_copy));
+
+        copy(arr_begin, arr_end, arr_copy_begin);
+        assert(equal(arr_begin, arr_end, arr_copy_begin, arr_copy_end));
+        fill(arr_copy_begin, arr_copy_end, 0);
+        assert(count(arr_copy_begin, arr_copy_end, 0) == arr_size);
+
+        ranges::copy(arr_begin, arr_end, arr_copy_begin);
+        assert(ranges::equal(arr_begin, arr_end, arr_copy_begin, arr_copy_end));
+        ranges::fill(arr_copy_begin, arr_copy_end, 0);
+        assert(ranges::count(arr_copy_begin, arr_copy_end, 0) == arr_size);
+
+        copy_n(arr_begin, arr_size, arr_copy_begin);
+        assert(equal(arr_begin, arr_end, arr_copy_begin, arr_copy_end));
+        fill_n(arr_copy_begin, arr_size, 0);
+        assert(count(arr_copy_begin, arr_copy_end, 0) == arr_size);
+
+        ranges::copy_n(arr_begin, arr_size, arr_copy_begin);
+        assert(ranges::equal(arr_begin, arr_end, arr_copy_begin, arr_copy_end));
+        ranges::fill_n(arr_copy_begin, arr_size, 0);
+        assert(ranges::count(arr_copy_begin, arr_copy_end, 0) == arr_size);
+    }
+
+    {
+        int arr_src[arr_size];
+        picky_contiguous_iterator arr_src_begin(begin(arr_src));
+        picky_contiguous_iterator arr_src_end(end(arr_src));
+
+        int arr_dest[arr_size] = {};
+        picky_contiguous_iterator arr_dest_begin(begin(arr_dest));
+        picky_contiguous_iterator arr_dest_end(end(arr_dest));
+
+        int remove_expected[] = {200, 210, 220, 240, 270, 280, 290, 300, 310, 320, 340, 370, 380, 390};

```

Comment:
For these "sub-tests", can you add braces when possible to scope the additional variables like `remove_expected` to a smaller region? This helps defend against copy-paste errors.